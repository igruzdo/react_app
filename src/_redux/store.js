import { combineReducers, createStore, applyMiddleware } from "redux";
import { messagesReducer } from './reducers/messagesReducer';
import { chatsListReducer } from './reducers/chatsListReducer';
import { nytReducer } from "./reducers/nytReducer";
import ReduxThunk from 'redux-thunk'
import { authReducer } from "./reducers/authReducer";
import * as messagesActions from "./actions/actionsTypes/messageActionsTypes"


// const robotAnswer = store => (dispatch, getState) => action => {

//     dispatch(action)

//     if(action?.type === messagesActions.ADD_MESSAGE) {
//         const timeout = setTimeout(() => {
//             dispatch({ type: messagesActions.ADD_MESSAGE, payload: { 
//                 name: 'Robot', 
//                 message: `I receved your message, ${action.payload.name}`, 
//                 date: new Date().toLocaleString(), 
//                 isRobot: true, 
//                 key: action.payload.key + 1, 
//                 chatId: action.payload.chatId
//             }})
//         }, 1500);

//         return () => {
//             clearTimeout(timeout)
//         }
//     }
// }

const reducer = combineReducers({
    messages: messagesReducer,
    chats: chatsListReducer,
    nytNews: nytReducer,
    auth: authReducer,
})

export const store = createStore(reducer, applyMiddleware(ReduxThunk));