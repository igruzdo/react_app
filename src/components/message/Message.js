import { useSelector } from 'react-redux'
import './message.css'

const Message = ({name, message, date }) => {

    const user = useSelector((state) => state.auth.user)

    let messageClassName = 'message_container'
    let positionClassName = 'position'

    if(user?.displayName !== name) {
        messageClassName += ' robot'
        positionClassName = ' robot_position'
    }

    return (
        <div className={positionClassName}>
            <div className={messageClassName}>
                <div className='message_header'>
                    <h4 className='author'>{name}</h4>
                    <p className='date'>{date}</p>
                </div>
                <p className='message_txt'>{message}</p>
            </div>
        </div>
    )
}

export default Message;