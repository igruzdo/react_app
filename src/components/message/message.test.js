import renderer from 'react-test-renderer';
import Message from './message';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../_redux/store'

it('render message', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Message name="John" message="test message" date={6576547654}></Message>
        </Provider>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})