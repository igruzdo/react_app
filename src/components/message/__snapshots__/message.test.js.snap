// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render message 1`] = `
<div
  className=" robot_position"
>
  <div
    className="message_container robot"
  >
    <div
      className="message_header"
    >
      <h4
        className="author"
      >
        John
      </h4>
      <p
        className="date"
      >
        6576547654
      </p>
    </div>
    <p
      className="message_txt"
    >
      test message
    </p>
  </div>
</div>
`;
